test(ghcr-rust-compiler): fix image rust build openssl-src v111.25.1

## Reproduction

On the upstream main branch do this:

`DOCKER_BUILDKIT=1 docker build -f ./tools/docker/rust-compiler/Dockerfile . --tag crc`

Then watch it burn like this:

```sh
#8 105.3    Compiling openssl-src v111.25.1+1.1.1t
#8 105.5 error[E0599]: the method `join` exists for struct `Vec<&OsStr>`, but its trait bounds were not satisfied
#8 105.5    --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/openssl-src-111.25.1+1.1.1t/src/lib.rs:335:55
#8 105.5     |
#8 105.5 335 |                     ar.get_args().collect::<Vec<_>>().join(OsStr::new(" ")),
#8 105.5     |                                                       ^^^^ method cannot be called on `Vec<&OsStr>` due to unsatisfied trait bounds
#8 105.5     |
#8 105.5     = note: the following trait bounds were not satisfied:
#8 105.5             `[&OsStr]: Join<_>`
#8 105.5 
#8 105.5 error[E0599]: the method `join` exists for struct `Vec<&OsStr>`, but its trait bounds were not satisfied
#8 105.5    --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/openssl-src-111.25.1+1.1.1t/src/lib.rs:344:59
#8 105.5     |
#8 105.5 344 |                     ranlib.get_args().collect::<Vec<_>>().join(OsStr::new(" ")),
#8 105.5     |                                                           ^^^^ method cannot be called on `Vec<&OsStr>` due to unsatisfied trait bounds
#8 105.5     |
#8 105.5     = note: the following trait bounds were not satisfied:
#8 105.5             `[&OsStr]: Join<_>`
#8 105.5 
#8 105.5 For more information about this error, try `rustc --explain E0599`.
#8 105.5 error: could not compile `openssl-src` due to 2 previous errors
#8 105.5 warning: build failed, waiting for other jobs to finish...
#8 109.6 error: failed to compile `wasm-pack v0.10.3`, intermediate artifacts can be found at `/tmp/cargo-install5DYG4N`
#8 109.6 
#8 109.6 Caused by:
#8 109.6   build failed
#8 ERROR: executor failed running [/bin/sh -c cargo install wasm-pack]: exit code: 101
------
 > [4/4] RUN cargo install wasm-pack:
------
executor failed running [/bin/sh -c cargo install wasm-pack]: exit code: 101
```

## Proposed Fix

Upgrade the Rust compiler version in the image from whatever it is the current latest stable so that the libraries that had their source code updated in the meantime can be compiled without issues.

<!-- Edit the body of your new issue then click the ✓ "Create Issue" button in the top right of the editor. The first line will be the issue title. Assignees and Labels follow after a blank line. Leave an empty line before beginning the body of the issue. -->

--- a/tools/docker/rust-compiler/Dockerfile
+++ b/tools/docker/rust-compiler/Dockerfile
@@ -1,4 +1,4 @@
-FROM rust:1.57.0-slim-bullseye
+FROM rust:1.68.0-slim-bullseye
 
 RUN apt update
 

